src/sentry/workflow_engine/models/workflow_fire_history.py

```diff
@@ -0,0 +1,21 @@
+from django.db.models import Index
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import CharField, FlexibleForeignKey, region_silo_model, sane_repr
+from sentry.db.models.base import DefaultFieldsModel
+
+
+@region_silo_model
+class WorkflowFireHistory(DefaultFieldsModel):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    workflow = FlexibleForeignKey("workflow_engine.Workflow", db_constraint=False)
+    group = FlexibleForeignKey("sentry.Group", db_constraint=False)
+    event_id = CharField("event_id", max_length=32, null=True)
+
+    class Meta:
+        db_table = "workflow_engine_workflowfirehistory"
+        app_label = "workflow_engine"
+        indexes = [Index(fields=["workflow", "date_added"])]
```

Comment:
keeping this index from `RuleFireHistory` because we are still supporting `lastTriggered` in the new UI https://github.com/getsentry/sentry/blob/796a0d1fc11ffd7d3cb2c4dd79d145fc356e18a4/src/sentry/api/serializers/models/rule.py#L184-L190

Reply:
Tbh this index only kind of helps with this query. If we wanted to make it fast, we could do something like

```
result_qs = reduce(
    lambda q1, q2: q1.union(q2),
    [RuleFireHistory.objects.filter(rule=item)
       .order_by('-date_added')
       .values('rule_id', 'date_added')[:1]
     for item in item_list]
)
```

The problem is that postgres isn't able to optimize the query with the IN, and so it does a bunch of sorting in memory. Since we have only a few rows in serializers, it'd be quicker to use UNION, which allows each individual query to be fast.